app/services/GeoServer.js

```diff
@@ -29,6 +29,66 @@ export async function createChallenge(challenge, successCallback, failureCallbac
   }
 }
 
+
+export async function getChallenges() {
+  let formData = new FormData();
+  formData.append('token', 'geo-ninjas');
+  formData.append('user', 2);  //for now
+
+  try{
+    const response = await fetch('http://enexia.com:10000/geo-challenge/challenge/search', {
+      method: 'POST',
+      body: formData
+    })
+    const responseJson = await response.json();
+    if(responseJson.success) {
+      return responseJson.challenges;
+    } else {
+      return undefined;
+    }
+  } catch (e){
+    failureCallback(e);
+  }
+}
+
+
+export async function getCompleted() {
+  let formData = new FormData();
+  formData.append('token', 'geo-ninjas');
+  formData.append('user', 2); //for now
+
+  try{
+    const response = await fetch('http://enexia.com:10000/geo-challenge/achievement/getAllByUser', {
+      method: 'POST',
+      body: formData
+    })
+    const responseJson = await response.json();
+    if(responseJson.success) {
+      var completedChallenges = [];
+      responseJson.achievements.forEach(async (achievement)=>{
+        let getChallengeData = new FormData();
+        getChallengeData.append('token', 'geo-ninjas');
+        getChallengeData.append('challenge', achievement.challenge);
+        const getChallResponse = await fetch('http://enexia.com:10000/geo-challenge/challenge/get', {
+          method: 'POST',
+          body: getChallengeData
+        })
+        const getChallResponseJson = await getChallResponse.json();
+        if(getChallResponseJson.success) {
+          completedChallenges.push(getChallResponseJson.challenge);
+        }
```

Comment:
What happens if the challenge request had an error? you might want to have a fallback that handles that.